scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -0,0 +1,237 @@
+use anyhow::anyhow;
+use async_trait::async_trait;
+use itertools::Itertools;
+use scylla::frame::response::result::CqlValue;
+use scylla::prepared_statement::PreparedStatement;
+use scylla::query::Query;
+use scylla::transport::topology::{ColumnKind, CqlType, Table};
+use scylla::Session;
+use scylla_cdc::consumer::*;
+use std::slice::Iter;
+use std::sync::Arc;
+
+struct ReplicatorConsumer {
+    dest_session: Arc<Session>,
+    dest_keyspace_name: String,
+    dest_table_name: String,
+    // [TODO]: Not sure, if we want to keep the ownership here,
+    //   or just a ref and keep it in the factory.
```

Comment:
In this case, it's better to either keep the table description by value (`Table`) or via a thread-safe shared pointer (`Arc<Table>`). Our code will spawn multiple consumers and run them on different tokio tasks. It's not really possible to have references pointing "between" tokio tasks.